pkg/client/unversioned/clientcmd/config.go

```diff
@@ -0,0 +1,421 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clientcmd
+
+import (
+	"errors"
+	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+)
+
+// ConfigAccess is used by subcommands and methods in this package to load and modify the appropriate config files
+type ConfigAccess interface {
```

Comment:
Can't `ClientConfigLoadingRules` implement this interface?

```go
GetLoadingPrecedence -> return Precedence
GetStartingConfig() -> call through to NewNonInteractiveDeferredLoadingClientConfig with no overrides?
GetDefaultFilename -> explicit if you have one, otherwise first existing from precedence, otherwise first from precedence
IsExplicitFilename -> len(explicitpath)
GetExplicitFile -> explicitPath
```

Seems like that about covers it and would eliminate the duplicate information.